contracts/examples/fractional-nfts/src/fractional_uri_info.rs

```diff
@@ -0,0 +1,35 @@
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+use elrond_wasm::contract_base::ManagedSerializer;
+
+#[derive(TopEncode, TopDecode)]
+pub struct FractionalUriInfo<M: ManagedTypeApi> {
+    pub original_payment: EsdtTokenPayment<M>,
+    pub initial_fractional_amount: BigUint<M>,
+}
+
+impl<M: ManagedTypeApi> FractionalUriInfo<M> {
+    pub fn new(
+        original_payment: EsdtTokenPayment<M>,
+        initial_fractional_amount: BigUint<M>,
+    ) -> Self {
+        Self {
+            original_payment,
+            initial_fractional_amount,
+        }
+    }
+
+    pub fn from_uris(uris: ManagedVec<M, ManagedBuffer<M>>) -> Self {
+        let first_uri = uris.get(0);
```

Comment:
I would add some more explicit error handling here, otherwise the error messages might be hard to understand.

`uris.try_get(0).unwrap_or_else(|| sc_panic!(...))`

Reply:
Done. I had to use `M::error_api_impl().signal_error` instead of `sc_panic!` since the latter uses `Self::Api` which is available only in the main contract.